doc/api-management.md

```diff
@@ -0,0 +1,145 @@
+## Stability
+
+Conceptually, an API is all assumptions adopters depend on to make use or extend Theia.
+A stable API is based on assumptions that are not subject to change.
+It does not matter whether an API is public or internal, or whether it is used a lot or never.
+
+> For instance, the language server protocol depends on such data types like URIs and positions
+since they don't change from language to language. It ensures its stability.
+
+API stability is indicated explicitly by adding `@experimental` or `@stable` js-doc tags.
+
+The explicit stability tag should be accompanied by `@since` tag indicating when API was added.
+The `@stable` tag should mention since which version an API was finalized.
+
+An API without a stability tag is considered to be experimental and does not require `@since` tag.
+
+```ts
+/**
+ * One does not need any annotations while working on experimental APIs.
+ */
+export interface ExperimentalInterface {
+}
+
+/**
+ * @since 0.1.0
+ * @stable since 1.0.0
+ */
+export interface StableInterface {
+
+    /**
+     * The same as `StableInterface`.
+     */
+    stableMethod(): void;
+
+    /**
+     * Adding new API to stable API should be explicit.
+     * 
+     * @since 1.1.0
+     * @experimental
+     */
+    experimentalMethod(): void;
+
+}
+
+```
+
+### Experimental
+
+- All new APIs should always be added as **experimental** since it's almost impossible to get [stable API](#stable) right first time.
+- Experimental APIs don't require the stability tag, but if a new member is added to [stable API](#stable) then it should be explicitly annotated.
+- Experimental APIs don't require extensive documentation. It does not mean that one should not document not obvious parts.
+- Experimental APIs don't follow [semver](https://semver.org/#spec-item-8) semantic.
+- Experimental APIs can be changed or removed without [the deprecation cycle](#deprecation) if they were not widely adopted.
+- Adoption should be measured by the number of internal clients or based on the feedback of Theia contributors and committers.
+
+### Stable
+
+- [Experimental APIs](#experimental) can be graduated to `stable` via [the finalization cycle](#finalization).
+- Stable APIs should be based on design decisions that are not likely to change.
+- Stable APIs should have sufficient adoption.
+- Stable APIs should have proper documentation.
+- Stable APIs must follow [semver](https://semver.org/#spec-item-8) semantic.
+- They can be changed only in a backward-compatible fashion.
+- They can be removed only via [the deprecation cycle](#deprecation).
+
+## Finalization
+
+API finalization should be requested via GitHub issue with `api-finalization` label.
+
+Finalization implies a review of the adoption, stability, and documentation of APIs.
+One cannot judge stability without gaining enough experience
+and provide stable documentation without API stability in the first place.
+
+If there is not enough adoption then finalization should be postponed.
```

Comment:
`If there is not enough adoption then` -> `If there is not enough adoption, then`